rapids_dask_dependency/dask_loader.py

```diff
@@ -3,65 +3,60 @@
 import importlib
 import importlib.abc
 import importlib.machinery
+import importlib.util
 import sys
-import warnings
 from contextlib import contextmanager
 
-from .patches.dask import patches as dask_patches
-from .patches.distributed import patches as distributed_patches
-
-original_warn = warnings.warn
-
-
-def _warning_with_increased_stacklevel(
-    message, category=None, stacklevel=1, source=None, **kwargs
-):
-    # Patch warnings to have the right stacklevel
-    # Add 3 to the stacklevel to account for the 3 extra frames added by the loader: one
-    # in this warnings function, one in the actual loader, and one in the importlib
-    # call (not including all internal frames).
-    original_warn(message, category, stacklevel + 3, source, **kwargs)
-
-
-@contextmanager
-def patch_warning_stacklevel():
-    warnings.warn = _warning_with_increased_stacklevel
-    yield
-    warnings.warn = original_warn
+from rapids_dask_dependency.utils import patch_warning_stacklevel, update_spec
 
 
 class DaskLoader(importlib.abc.MetaPathFinder, importlib.abc.Loader):
+    def __init__(self):
+        self._blocklist = set()
+
     def create_module(self, spec):
         if spec.name.startswith("dask") or spec.name.startswith("distributed"):
-            with self.disable(), patch_warning_stacklevel():
-                mod = importlib.import_module(spec.name)
+            with self.disable(spec.name):
+                try:
```

Comment:
Do you have a sense for how much overhead it is to try importing a patch for every module beginning with "dask" or "distributed"? Perhaps we should register the patches somehow so we don't even bother trying unless we expect the import to succeed? For example, maybe the `patches` directory can contain a `__init__.py` file with an `_rdd_patches = [...]` list of modules that are patched? (feel free to ignore this thought/suggestion for now - especially if the current overhead is low)

Reply:
It's lost in the noise of import.

Reply:
Okay, fair enough.

Reply:
Yeah I did some quick measurements and it's noise. Importing dask is slow already.